XenAdmin/Dialogs/CallHome/CallHomeSettingsDialog.cs

```diff
@@ -46,12 +46,16 @@ public partial class CallHomeSettingsDialog : XenDialogBase
         private bool authenticationRequired;
         private bool authenticated;
         private string authenticationToken;
+        private string usetName;
+        private string password;
 
         public CallHomeSettingsDialog(Pool pool)
         {
             this.pool = pool;
-            callHomeSettings = pool.CallHomeSettings;
-            authenticationToken = callHomeSettings.GetExistingUploadToken(pool.Connection);
+            callHomeSettings = pool.CallHomeSettings;
+            authenticationToken = callHomeSettings.GetExistingSecretyInfo(pool.Connection, CallHomeSettings.UPLOAD_TOKEN_SECRET);
+            usetName = callHomeSettings.GetExistingSecretyInfo(pool.Connection, CallHomeSettings.UPLOAD_CREDENTIAL_USER_SECRET);

```

Comment:
For userName and Password, we need to call GetSecretyInfo so that we get the value from the settings of this connection only. GetExistingSecretyInfo will also try to get the values from other connections, which shouldn't be done this case.